_search-plugins/star-tree-index.md

```diff
@@ -0,0 +1,178 @@
+- Once Star Tree index is enabled for an index, it cannot be disabled. The data has to be re-indexed without the Star Tree mapping to remove Star Tree from the index.
+    - Changing Star Tree configuration will also require a re-index operation.
+- [Multi-values/array values]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/index/#arrays) are not supported
+- Only [limited queries and aggregations](#supported-query-and-aggregations) are supported currently. Support for more features will be released in upcoming versions.
+- The cardinality of the dimensions should not be very high (like `_id` fields), otherwise it leads to high storage usage and higher query latency.
+
+## Enabling star tree index
+- Set the feature flag `opensearch.experimental.feature.composite_index.star_tree.enabled"` to `true`. For more information about enabling and disabling feature flags, see [Enabling experimental features]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/experimental/).
+- Set the `indices.composite_index.star_tree.enabled` setting to `true`. For instructions on how to configure OpenSearch, see [configuring settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#static-settings).
+- Set the `index.composite_index` index setting to `true` during index creation.
+
+## Examples
+
+The following examples show how to use Star Tree index.
+
+### Defining star tree index in mappings
+
+Define Star Tree configuration in index mappings when creating an index. <br/>
+To create Star Tree index to pre-compute aggregations for `request_size` and `latency` fields for all the combinations of values in `port` and `status` fields indexed in the `logs` index, configure the following mapping:
+
+```json
+PUT logs
+{
+  "settings": {
+    "index.number_of_shards": 1,
+    "index.number_of_replicas": 0,
+    "index.composite_index": true
+  },
+  "mappings": {
+    "composite": {
+      "startree1": {
+        "type": "star_tree",
+        "config": {
+          "ordered_dimensions": [
+            {
+              "name": "status"
+            },
+            {
+              "name": "port"
+            }
+          ],
+          "metrics": [
+            {
+              "name": "size",
+              "stats": [
+                "sum"
+              ]
+            },
+            {
+              "name": "latency",
+              "stats": [
+                "avg"
+              ]
+            }
+          ]
+        }
+      }
+    },
+    "properties": {
+      "status": {
+        "type": "integer"
+      },
+      "port": {
+        "type": "integer"
+      },
+      "size": {
+        "type": "integer"
+      },
+      "latency": {
+        "type": "scaled_float",
+        "scaling_factor": 10
+      }
```

Comment:
Mapping for property `request_size` is missing here and it will cause the search queries to fail with NullPointerException. 
```
      "request_size": {
        "type": "integer"
      }
```